src/server/qtquick/wsurfaceitem.cpp

```diff
@@ -530,6 +530,31 @@ void WSurfaceItem::setRightPadding(qreal newRightPadding)
     Q_EMIT rightPaddingChanged();
 }
 
+qreal WSurfaceItem::surfaceSizeRatio() const
+{
+    Q_D(const WSurfaceItem);
+    return d->surfaceSizeRatio;
+}
+
+void WSurfaceItem::setSurfaceSizeRatio(qreal ssr)
+{
+    Q_D(WSurfaceItem);
+
+    if (qFuzzyCompare(ssr, d->surfaceSizeRatio))
+        return;
+    d->surfaceSizeRatio = ssr;
+    Q_EMIT surfaceSizeRatioChanged();
+
+    surfaceSizeRatioChange();
```

Comment:
```
Q_EMIT surfaceSizeRatioChanged();
 surfaceSizeRatioChange();
```
?

Reply:
`surfaceSizeRatioChanged` is a signal to notify this property changed.
`surfaceSizeRatioChange` is a virtual function to allow the derived class do someting when surface size ratio changed.